markets/perps-market/contracts/storage/AsyncOrder.sol

```diff
@@ -199,7 +200,22 @@ library AsyncOrder {
         // TODO: validate position size
         oldPosition = PerpsMarket.load(order.marketId).positions[order.accountId];
 
+        if (
+            PerpsMarket.validatePositionSize(
+                perpsMarketData,
+                marketConfig.maxMarketValue,
+                oldPosition.size,
+                order.sizeDelta
+            )
```

Comment:
let the `validatePositionSize` just revert instead of it returning a bool.  

that function needs to be refactored; but first would write the tests and make sure its working properly.